kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/PaginatedScanSuite.scala

```diff
@@ -0,0 +1,268 @@
+import java.util.Optional
+
+import io.delta.kernel.data.FilteredColumnarBatch
+import io.delta.kernel.defaults.DeltaTableWriteSuiteBase
+import io.delta.kernel.defaults.engine.{DefaultEngine, DefaultJsonHandler, DefaultParquetHandler}
+import io.delta.kernel.defaults.utils.{ExpressionTestUtils, TestUtils}
+import io.delta.kernel.internal.PaginatedScanImpl
+import io.delta.kernel.internal.replay.{PageToken, PaginatedScanFilesIteratorImpl}
+
+import org.apache.spark.sql.delta.DeltaLog
+
+import org.apache.spark.sql.catalyst.plans.SQLHelper
+import org.scalatest.funsuite.AnyFunSuite
+import org.slf4j.{Logger, LoggerFactory};
+
+class PaginatedScanSuite extends AnyFunSuite with TestUtils
+    with ExpressionTestUtils with SQLHelper with DeltaTableWriteSuiteBase {
+
+  private val logger = LoggerFactory.getLogger(classOf[PaginatedScanSuite])
+  // TODO: test call hasNext() twice
+
+  test("Read First Page: basic pagination with single JSON file") {
+    withTempDir { tempDir =>
+      /**
+       *   Creates a Delta table with 10 Parquet data files; each file contains 10 rows
+       * * 00000000000000000000.json contains 10 AddFile actions — one per file
+       */
+      spark.range(0, 100, 1, 10).write.format("delta").save(tempDir.getCanonicalPath)
+
+      val snapshot = latestSnapshot(tempDir.getCanonicalPath)
+      val pageSize = 8 // TODO: change different page sizes
+      val paginatedScan = snapshot.getScanBuilder
+        .buildPaginated(pageSize, Optional.empty() /* page token opt */ )
+        .asInstanceOf[PaginatedScanImpl]
+
+      /**
+       * Creates a custom engine with batch size 5;
+       * 00000000000000000000.json contains 3 batches, 10 active AddFiles in total
+       * Batch 1: 5 rows, 2 selected AddFiles
```

Comment:
You should explain why this is only 2 addfiles -- it is because of the CommitInfo, Protocol, and Metadata actions. I had to think for a seccond to realize this :) 

Could you even make this clearer in the code? e.g. what happens if we change how many non-addfile rows are written in the first commit? we want to make our test robust, not brittle, to change

```scala
val numNonAddFileRows = 3 // CommitInfo, Protocol, and Metadata
...
```